api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java

```diff
@@ -74,6 +86,29 @@ public void fetchFromNational(BiometricSubject nationalBiometricSubject, M2SysCa
         registerLocally(nationalBiometricSubject, fingerScan);
     }
 
+    @Override
+    public void synchronizeFingerprints(M2SysCaptureResponse fingerScan, FingerScanStatus fingerScanStatus) {
+        if (fingerScanStatus.isRegisteredNationally()) {
+            if (patientHelper.findByNationalFpId(fingerScanStatus.getNationalBiometricSubject().getSubjectId()) == null) {
+                attachNationalIdToThePatient(fingerScanStatus.getLocalBiometricSubject().getSubjectId());
```

Comment:
Why getLocalBiometricSubject? I think there should be getNationalBiometricSubject.

Reply:
You're right